### ISSUE TITLE
Guides - Create an upgrade path section

### ISSUE THREAD

**Issue by knittingcodemonkey** [webpack/webpack.js.org#1795]
Create an upgrade path section with clearly defined, individual pages:

- Upgrade from v1 to v2
- Upgrade from v2 to v3
- Upgrade from v3 to v4

Each page should include any script helpers, and examples particular to it's version of the upgrade.

**Comment by skipjack** [comment]
I like this idea as well however I want to lump it in with #1706 as I was already planning to do something similar. Will add your point as a comment. Also note that we excluded v2 - v3 intentionally as there weren't very many breaking changes (almost none for most users). However, we could add a small page that notes this and invites people to document the few breaking changes that did occur.

TLDR; closing... see #1706 for updates.
